Add info about conflict between Release and Debug TBB to Known Limitations (#1663)
uxlfoundation/oneTBB: doc/main/intro/limitations.rst

This page outlines the known limitations of oneTBB to help you better understand its capabilities. 

> Using Debug TBB in SYCL Program

Reviewer: ```suggestion
Debug TBB In The SYCL Program
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> **Limitation:** When using the Debug version of oneAPI Threading Building Blocks (TBB) in a SYCL program compiled using the Intel(R) oneAPI DPC++/C++ Compiler, the application may crash due to two conflicting ``tbb`` and ``tbb_debug`` libraries being loaded at the same time.

Author: ```suggestion
**Limitation:** When using the Debug version of oneAPI Threading Building Blocks (oneTBB) in a SYCL program compiled using the Intel(R) oneAPI DPC++/C++ Compiler, the application may crash due to two conflicting ``tbb`` and ``tbb_debug`` libraries being loaded at the same time.
```

> **Limitation:** When using the Debug version of oneAPI Threading Building Blocks (oneTBB) in a SYCL program compiled using the Intel(R) oneAPI DPC++/C++ Compiler, the application may crash due to two conflicting ``tbb`` and ``tbb_debug`` libraries being loaded at the same time.

Reviewer: ```suggestion
**Limitation:** The application may crash when using the Debug version of oneTBB in a SYCL program compiled with Intel(R) oneAPI DPC++/C++ Compiler. This happens because both ``tbb`` (Release version) and ``tbb_debug`` (Debug version) libraries load simultaneously, causing conflicts.

> **Solution:** Link the applictation with the Release version ``tbb`` or use ``qtbb`` flag provided by the Intel(R) oneAPI DPC++/C++ Compiler.

Reviewer: ```suggestion
**Solution:** Do one of the following:

* Link the application with the Release version ``tbb`` instead of ``tbb_debug``.
* Use the ``qtbb`` flag provided by the Intel(R) oneAPI DPC++/C++ Compiler.
```

This page outlines the known limitations of oneTBB to help you better understand its capabilities. 

Debug TBB In The SYCL Program
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> **Limitation:** The application may crash when using the Debug version of oneTBB in a SYCL program compiled with Intel(R) oneAPI DPC++/C++ Compiler. This happens because both ``tbb`` (Release version) and ``tbb_debug`` (Debug version) libraries load simultaneously, causing conflicts.

Reviewer 2: May be it is better to rephrase it - "when linking with the Debug oneTBB library"? Or is it clear what is using the Debug version of oneTBB?

Author: I suppose that "using ... in a SYCL program" implies thate the application is using the Debug TBB.
